gtsam/hybrid/CLGaussianConditional.h

```diff
@@ -0,0 +1,50 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   CLGaussianConditional.h
+ * @brief  A hybrid conditional in the Conditional Linear Gaussian scheme
+ * @author Fan Jiang
+ * @date   Mar 12, 2022
+ */
+
+#include <gtsam/inference/Conditional.h>
+#include <gtsam/hybrid/HybridFactor.h>
+
+#include <gtsam/linear/GaussianConditional.h>
+#include <gtsam/discrete/DecisionTree.h>
+
+namespace gtsam {
+class CLGaussianConditional : public HybridFactor, public Conditional<HybridFactor, CLGaussianConditional> {
```

Comment:
How is `CLGaussianConditional` different from `GaussianMixture`? They're both linear Gaussian and they are conditionals.

Reply:
CLGaussianConditional will have weights (see Murphy02 for CLG)

this is necessary because it is possible for the multifrontal elimination to ask for this:

```
x1 cont, c1 disc. now x1, c1 is a clique, so multifrontal will call eliminate with
ordering: x1, c1

thus we will need to give
cond: p(x1,c1)=p(x1|c1)p(c1)
```
which is not GM

Reply:
I think this is also the answer to what @dellaert has asked